language/command-line-common/src/testing.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::env::read_bool_env_var;
+
+/// Extension for raw output files
+pub const OUT_EXT: &str = "out";
+/// Extension for expected output files
+pub const EXP_EXT: &str = "exp";
+
+/// If any of these env vars is set, the test harness should overwrite
+/// the existing .exp files with the output instead of checking
+/// them against the output.
+pub const UPDATE_BASELINE: &str = "UPDATE_BASELINE";
+pub const UPBL: &str = "UPBL";
+pub const UB: &str = "UB";
+
+pub fn read_env_update_baseline() -> bool {
+    read_bool_env_var(UPDATE_BASELINE) || read_bool_env_var(UPBL) || read_bool_env_var(UB)
+}
+
+pub fn format_diff(expected: String, actual: String) -> String {
+    use difference::*;
+
+    let changeset = Changeset::new(&expected, &actual, "\n");
+
+    let mut ret = String::new();
+
+    for seq in changeset.diffs {
+        match &seq {
+            Difference::Same(x) => {
+                ret.push_str(x);
+                ret.push('\n');
+            }
+            Difference::Add(x) => {
+                ret.push_str("\x1B[92m");
+                ret.push_str(x);
+                ret.push_str("\x1B[0m");
+                ret.push('\n');
+            }
+            Difference::Rem(x) => {
+                ret.push_str("\x1B[91m");
+                ret.push_str(x);
+                ret.push_str("\x1B[0m");
+                ret.push('\n');
```

Comment:
I wonder if we should replace these color codes with some cross platform crates like [termcolor](https://docs.rs/termcolor/1.1.2/termcolor/)

Reply:
We should, but a job for another day